### PR TITLE
Reciter page

### PR DESCRIPTION
**Pull request by muhajirdev**

### CODE REVIEW

**Comment by mmahalwy** [comment]
#### 📝 Changed routes:


| [**/reciter/\[reciterId\]**](https://quran-com-nayibx89f-qurancom.vercel.app/reciter/[reciterId]) | [**…ter/reciterPage.module**](https://quran-com-nayibx89f-qurancom.vercel.app/reciter/reciterPage.module) |
|:-:|:-:|
|<a href="https://quran-com-nayibx89f-qurancom.vercel.app/reciter/[reciterId]">Screenshot of /reciter/[reciterId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/pNubhLqzYRY3VfRXINCElDabio6rXl0mW4mlwjDs0iHandYURey_-iozCG6UjYUmtOe6_7SGLWEd2BwUyR0NMA.png">(view full size)</a> |<a href="https://quran-com-nayibx89f-qurancom.vercel.app/reciter/reciterPage.module">Screenshot of /reciter/reciterPage.module</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/7_dTQBo4T5axU8efENCGjbzQcrEmxa8rw1Xzb7HwwxFQYgjqB6G4Ks3yNYeOwnvGCd5QEkUiuZLwjsQ5kD1O7A.png">(view full size)</a> |







Commit e4d93856f32963fc3a1b7e491ce8f4be5ced1c36 (https://quran-com-nayibx89f-qurancom.vercel.app).

File: src/components/Radio/ReciterStationList.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ import Reciter from 'types/Reciter';
 // TODO:
 // - these images url should come from backend.
 // - find better images
-const reciterPictures = {
+export const reciterPictures = {
```
Comment (muhajirdev) [inline src/components/Radio/ReciterStationList.tsx:24]: will create another PR to remove this const

File: src/components/Reciter/ReciterInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import styles from './ReciterInfo.module.scss';
+
+import { playFrom } from 'src/redux/slices/AudioPlayer/state';
+import { setRadioStationState } from 'src/redux/slices/radio';
+import { getImageCDNPath } from 'src/utils/api';
+import { getRandomChapterId } from 'src/utils/chapter';
+import { logEvent } from 'src/utils/eventLogger';
+import Reciter from 'types/Reciter';
+
+const BIO_SAMPLE =
```
Comment (muhajirdev) [inline src/components/Reciter/ReciterInfo.tsx:18]: currently bio is a bit too long, and using html. using this for a preview for now

File: src/components/Reciter/ChapterList.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+import { playFrom } from 'src/redux/slices/AudioPlayer/state';
+import { setRadioStationState } from 'src/redux/slices/radio';
+import { getRandomChapterId } from 'src/utils/chapter';
+import { logEvent } from 'src/utils/eventLogger';
+import { getWindowOrigin } from 'src/utils/url';
+// import Chapter from 'types/Chapter';
+import Reciter from 'types/Reciter';
+
+type ChapterListProps = {
+  filteredChapters: any;
```
Comment (muhajirdev) [inline src/components/Reciter/ChapterList.tsx:27]: fixing

File: src/components/Reciter/ChapterList.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+import { getChapterAudioData } from 'src/api';
+import { playFrom } from 'src/redux/slices/AudioPlayer/state';
+import { getRandomChapterId } from 'src/utils/chapter';
+import { logEvent } from 'src/utils/eventLogger';
+import { getWindowOrigin } from 'src/utils/url';
+// import Chapter from 'types/Chapter';
+import Chapter from 'types/Chapter';
+import Reciter from 'types/Reciter';
+
+type ChapterListProps = {
```
Comment (tifa2UP) [inline src/components/Reciter/ChapterList.tsx:29]: ChaptersList

File: src/components/Reciter/ChapterList.tsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+import { ToastStatus, useToast } from '../dls/Toast/Toast';
+
+import styles from './ChapterList.module.scss';
+
+import { getChapterAudioData } from 'src/api';
+import { playFrom } from 'src/redux/slices/AudioPlayer/state';
+import { getRandomChapterId } from 'src/utils/chapter';
+import { logEvent } from 'src/utils/eventLogger';
+import { getWindowOrigin } from 'src/utils/url';
+// import Chapter from 'types/Chapter';
```
Comment (tifa2UP) [inline src/components/Reciter/ChapterList.tsx:25]: remove

File: src/components/Reciter/ChapterList.tsx [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+import Chapter from 'types/Chapter';
+import Reciter from 'types/Reciter';
+
+type ChapterListProps = {
+  filteredChapters: Chapter[];
+  selectedReciter: Reciter;
+};
+
+const ChapterList = ({ filteredChapters, selectedReciter }: ChapterListProps) => {
```
Comment (tifa2UP) [inline src/components/Reciter/ChapterList.tsx:34]: ditto

File: src/components/Reciter/ChapterList.tsx [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+  const toast = useToast();
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+
+  const [downloadingChapterAudio, setDownloadingChapterAudio] = useState(null);
```
Comment (tifa2UP) [inline src/components/Reciter/ChapterList.tsx:39]: CurrentlyDownloadingChapterAudioId

File: src/components/Reciter/ChapterList.tsx [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+  const onPlayChapter = (chapterId?: string) => {
+    const selectedChapterId = chapterId || getRandomChapterId().toString();
```
Comment (tifa2UP) [inline src/components/Reciter/ChapterList.tsx:42]: remove random part. Make chapter required

File: src/components/Reciter/ChapterList.tsx [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+            <Button
+              variant={ButtonVariant.Ghost}
+              size={ButtonSize.Small}
+              shape={ButtonShape.Circle}
+              onClick={(e) => {
+                e.stopPropagation();
+                const origin = getWindowOrigin();
+                clipboardCopy(`${origin}/${chapter.id}`).then(() => {
+                  toast(t('common:shared'), { status: ToastStatus.Success });
+                });
```
Comment (tifa2UP) [inline src/components/Reciter/ChapterList.tsx:92]: define the functions separately. OnCopyUrlClicked

File: src/components/Reciter/ChapterList.tsx [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+              }}
+            >
+              <CopyLinkIcon />
+            </Button>
+            <Button
+              shape={ButtonShape.Circle}
+              variant={ButtonVariant.Ghost}
+              size={ButtonSize.Small}
+              onClick={async (e) => {
```
Comment (tifa2UP) [inline src/components/Reciter/ChapterList.tsx:101]: ditto

File: src/components/Reciter/ChapterList.tsx (hunk 9 above)
Comment (tifa2UP) [reply src/components/Reciter/ChapterList.tsx:92]: log

File: src/components/Reciter/ReciterInfo.tsx [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+  'Lorem ipsum dolor sit amet consectetur adipisicing elit. Rerum, id quia. Optio reiciendis officia nobis accusantium quidem cumque, accusamus laboriosam amet officiis mollitia qui saepe, possimus, rem unde maxime atque?';
+
+type ReciterInfoProps = {
+  selectedReciter: Reciter;
+};
+const ReciterInfo = ({ selectedReciter }: ReciterInfoProps) => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+
+  const onPlayReciterStation = () => {
```
Comment (tifa2UP) [inline src/components/Reciter/ReciterInfo.tsx:28]: Do not duplicate radio functionality, should reuse the existing one

File: src/pages/reciters/[reciterId].tsx [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+  return resultItems as Chapter[];
+};
+
+type ReciterPageProps = { selectedReciter: Reciter };
+const Reciterpage = ({ selectedReciter }: ReciterPageProps) => {
+  const allChapterData = getAllChaptersData();
+  const router = useRouter();
+  const { t, lang } = useTranslation();
+
+  const reciterId = router.query.reciterId as string;
```
Comment (tifa2UP) [inline src/pages/reciters/[reciterId].tsx:54]: reciterId should be passed through the props, not through the router. Check how we do the chapter page Isr

File: src/pages/reciters/[reciterId].tsx [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const allChaptersWithId = useMemo(
```
Comment (tifa2UP) [inline src/pages/reciters/[reciterId].tsx:58]: add a comment explaining what this does

File: src/pages/reciters/[reciterId].tsx [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+
+export const getStaticProps: GetStaticProps = async ({ params }) => {
+  try {
+    const reciterId = params.reciterId as string;
+
+    const reciterData = await getReciterData(reciterId);
+
+    return {
+      props: {
+        selectedReciter: reciterData.reciter,
```
Comment (tifa2UP) [inline src/pages/reciters/[reciterId].tsx:119]: pass reciterId through here

**Review by tifa2UP — APPROVED** [review]

**Pull request merged**